### DISCUSSION TITLE
Feature: One to One Relationship

### DISCUSSION THREAD

**Discussion by r0b78** [directus/directus#2704] [Feature Requests]
The one to one relationship will be the same as o2m but with restriction of only one element, this will be a huge benefit for decouple tables and have better sql structure. Also the UI will give better UX and more naturally 

Show Case 
The main table coustumers 
![image](https://user-images.githubusercontent.com/22716876/81610134-3e6e1280-9396-11ea-93aa-f00f89bb714b.png)
You want to decouple and have customers and detail_custumers for business reasons.
The detail_customers table has this 3 fields description,tag and source 
![image](https://user-images.githubusercontent.com/22716876/81610344-94db5100-9396-11ea-85cf-5e7dc8d5b0a5.png)
When you go to add a new customer the have the fields of customers and detail_customers. The IU  is see it as a hole.
![image](https://user-images.githubusercontent.com/22716876/81610561-f996ab80-9396-11ea-8216-ca103bbf54d6.png)

**Comment by rijkvanzanten** [comment]
What will it show when there's no related item available? What if the one item is null? What happens when you add a new item? Should you be able to select a pre-existing item in case the link hasn't been made yet?

**Comment by dimitrov-adrian** [comment]
There are some use cases for such relation and functionality. Also in meaning of Directus, it will be ease way to implement group of properties for a record, as the repeater is a repeater in mind, and stores data in object, but this will open the drawer and will store data natively in the database.

In fact there may be not a correct answer what data to return in the item if no record is related - null, default values or null valued columns. May be nulled columns will be more naturally because the outer join returns null values.

Anyway, I think such functionality will be very powerful for data modeing, but it also could be abused and used only for visually data separation.

**Reply by benhaynes** [reply]
The idea of using this to organize fields is interesting, "open fields in drawer" could be an option of the future `group` interfaces.

**Reply by dimitrov-adrian** [reply]
Well group interface will works, if it not follows the repeater behvaviour and stores into JSON object. I am imaging one-to-one pretty much same to one-to-many, but for the interface to have just a button that open the relation drawer (no add existing, no create new, just button)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
